cpp/mcap/include/mcap/visibility.hpp

```diff
@@ -16,9 +17,18 @@
 #else
 #  define MCAP_EXPORT __attribute__((visibility("default")))
 #  define MCAP_IMPORT
+#  define MCAP_LOCAL __attribute__((visibility("hidden")))
 #  if __GNUC__ >= 4
 #    define MCAP_PUBLIC __attribute__((visibility("default")))
 #  else
 #    define MCAP_PUBLIC
 #  endif
 #endif
+
+#ifndef MCAP_VISIBILITY
+#  ifdef MCAP_VISIBILITY_LOCAL
```

Comment:
Is this a typo? I don't see MCAP_VISIBILITY_LOCAL anywhere else. Is the point that a user would define this themselves? If so, an explanation should probably also be added to the C++ readme at https://github.com/foxglove/mcap/blob/main/cpp/README.md

Is it common to provide this visibility option in this way? Do you have references to other popular projects that do something like this?